pubsub/api/Pubsub.Samples/ExactlyOnceDeliverySubscriberAsync.cs

```diff
@@ -0,0 +1,94 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_subscriber_exactly_once]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Google.Cloud.PubSub.V1.SubscriberClient;
+
+public class ExactlyOnceDeliverySubscriberAsyncSample
+{
+    public async Task<IEnumerable<string>> ExactlyOnceDeliverySubscriberAsync(string projectId, string subscriptionId)
+    {
+        // subscriptionId should be the ID of an exactly-once delivery subscription.
+        SubscriptionName subscriptionName = SubscriptionName.FromProjectSubscription(projectId, subscriptionId);
+        SubscriberClient subscriber = await SubscriberClient.CreateAsync(subscriptionName);
+        // To get the status of ACKnowledge (ACK) or Not ACKnowledge (NACK) request in exactly once delivery subscriptions,
+        // create a subscription handler that inherits from Google.Cloud.PubSub.V1.SubscriptionHandler. 
+        // For more information see Google.Cloud.PubSub.V1.SubscriptionHandler reference docs here:
+        // TODO: [add link to reference docs when we have them].
+        var subscriptionHandler = new SampleSubscriptionHandler();
+        Task subscriptionTask = subscriber.StartAsync(subscriptionHandler);
+        // The subscriber will be running until it is stopped.
+        await Task.Delay(5000);
+        await subscriber.StopAsync(CancellationToken.None);
+        // Let's make sure that the start task finished successfully after the call to stop.
+        await subscriptionTask;
+        return subscriptionHandler.SuccessfulAckedIds;
+    }
+
+    // Sample handler to handle messages and ACK/NACK responses.
+    public class SampleSubscriptionHandler : SubscriptionHandler
+    {
+        public ConcurrentBag<string> SuccessfulAckedIds { get; } = new ConcurrentBag<string>();
+
+        /// <summary>
+        /// The handler function that is passed all received messages.
+        /// This function may be called on multiple threads concurrently. Return <see cref="Reply.Ack"/> from this function
+        /// to ACKnowledge this message (implying it won't be received again); or return <see cref="Reply.Nack"/> to Not
+        /// ACKnowledge this message (implying it will be received again). If this function throws any Exception, then
+        /// it behaves as if it returned <see cref="Reply.Nack"/>.
```

Comment:
```suggestion
        /// The function that processes received messages. It should be thread-safe.
        /// Return <see cref="Reply.Ack"/> to ACKnowledge the message (meaning it won't be received again).
        /// Return <see cref="Reply.Nack"/> to Not ACKnowledge the message (meaning it will be received again).
        /// From the point of view of message akcnowledgement, throwing an exception is equivalent to returning see cref="Reply.Nack"/>.
```